R/ggplotMDS.R

```diff
@@ -529,11 +529,35 @@ MDS_var_explained <- function(mds,
         mds <- limma::plotMDS(mds, plot = FALSE)
     }
 
-    mds.distances <- mds %$% distance.matrix %>% as.dist
-    mdsvals <- mds.distances %>%
-        {suppressWarnings(cmdscale(., k = ncol(mds$distance.matrix) - 1))} %>%
-        magrittr::set_colnames(stringr::str_c("Dim", seq_len(ncol(.)))) %>%
-        as.data.frame
+    mds.distances <- tryCatch(
+        expr = {
+            mds %$% distance.matrix %>% as.dist
+        },
+        error = function(e) {
+            mds %$% distance.matrix.squared %>% as.dist
+        }
+    )
+
+    mdsvals <- tryCatch(
+        expr = {
+            mds.distances %>%
+                {
```

Comment:
Why you need to surround the statement with curly brackets though it is a single statement?

Reply:
not sure I was there before so I kept it. I tried removing them but it causes an error in the test. I did change the formatting so it would be hopefully a bit easier to read?